lib-opensource/emqx_management/src/emqx_mgmt_api_listeners.erl

```diff
@@ -41,6 +53,23 @@ list(_Binding, _Params) ->
     return({ok, [#{node => Node, listeners => format(Listeners)}
                               || {Node, Listeners} <- emqx_mgmt:list_listeners()]}).
 
+%% Restart listeners on a node.
+restart(#{node := Node, identifier := Identifier}, _Params) ->
+    case emqx_mgmt:restart_listener(Node, emqx_listeners:find_by_id(Identifier)) of
+        ok -> return({ok, "Listener restarted."});
+        {err, Error} -> return({err, Error})
+    end;
+
+%% Restart listeners in the cluster.
+restart(#{identifier := Identifier}, _Params) ->
+    Results = [emqx_mgmt:restart_listener(Node, emqx_listeners:find_by_id(Identifier)) || {Node, _Info} <- emqx_mgmt:list_nodes()],
```

Comment:
should not find listener locally in the caller node and pass it to a remote node,
the remote node should cal `find_by_id` by itself.